### PR TITLE
#530 | Redesign navbar

### LINKED ISSUE #530: Teloscan Homepage Redesign - Navbar
## Acceptance Criteria
The nav bar should have the following elements:
Top Bar:
- TLOS price - in the form of `TLOS Price: $0.00` (left)
- search bar (hidden if route is home)
- light/dark toggle

Bottom bar:
- Teloscan logo (left)
- Home
- Blockchain
    - Transactions
    - Blocks
- Developers
    - API Documentation
    - Verify Contract (Sourcify) (open in new tab)
- Telos Wallet (Link to Telos Wallet)
- More
    - Telos Ecosystem
    - CSV Export

On the bottom bar, the menu item corresponding to the route should be colored `primary`, e.g. when viewing the block detail page, the 'Blockchain' dropdown item should be primary color, while the rest are either black or white depending on dark mode

On scroll down, the bottom bar should hide. On scroll up, it should show

### PR DESCRIPTION
**Pull request by ezra-sg**
# Fixes #530 
## Description
This PR overhauls the navbar. it also adds the following changes:
- removes staking altogether
- adds new global variables to `app.scss`: 
    - `--grey-text-color` - muted text color (adapts to light/dark mode)
    - `--border-color` - border for use on tables and buttons etc (adapts to light/dark mode)
    - `--text-color` - default text color for when it needs to be explicitly stated (dark in light mode, white in dark mode)
- adds new high order button component, `OutlineButton`
- adds new file `src/lib/chain-utils.ts` for holding chain info until we switch over to true multi chain support

## Test scenarios
- go to https://deploy-preview-549--teloscan-redesign.netlify.app/
- check out the new header on different screen sizes
    - all elements should be accessible regardless of screen size (e.g. the Network button moves from the top right on desktop to the hamburger menu on mobile)
    - it should look good 😎 
    - bottom part of menu should go away on scroll
    - entire menu should be keyboard accessible
- log in using the connect wallet button
- check out the wallet menu by clicking the wallet button
    - you should be able to view TLOS balance, copy address, view account page, and logout
- change the language
    - the header text should all be localized
- checkout locally and run on **mainnet**
    - you should see TLOS balance fiat value in the account menu after logging in
    - you should see gas and TLOS price in the top left
    - you should not see the testnet indicator on the logo

## Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have cleaned up the code in the areas my change touches
-   [x] My changes generate no new warnings
-   [ ] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings
-   [x] I have removed any unnecessary console messages
-   [x] I have included all english text to the translation file and/or created a new issue with the required translations for the currently supported languages
-   [x] I have tested for mobile functionality and responsiveness
-   [ ] I have added appropriate test coverage

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -14,6 +28,48 @@ onMounted(() => {
...
+                return;
+            }
+            const loginObj = JSON.parse(loginData);
+            const wallet = ual.getAuthenticators().availableAuthenticators.find(a => a.getName() === loginObj.provider);
+            wallet?.logout();
+        }
+        $store.commit('login/setLogin', {});
+        localStorage.removeItem('loginData');
+        providerManager.setProvider(null);
+    });
```
Comment (ezra-sg) [inline src/App.vue:72]: unchanged, just moved away from the header component

File: src/components/LoginModal.vue [hunk 2]
```diff
@@ -215,78 +215,76 @@ function hideDialog(){
...
-                        class="c-login-modal__image-container"
-                        @click="connectBraveWallet()"
-                    >
-                        <q-img
-                            :src="require('src/assets/brave_lion.svg')"
-                            width="50px"
-                        />
-                        <span> Brave Wallet </span>
-                    </q-card>
+<q-dialog :model-value="props.show" @hide="hideDialog">
```
Comment (ezra-sg) [inline src/components/LoginModal.vue:218]: unchanged, just removed the wrapping `div`

File: src/components/header/AppHeaderWallet.vue [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+const prettySystemTokenBalanceFiat = computed(() => {
+    if (isTestnet()) {
+        return '';
+    }
+    const price = Number($store.getters['chain/tlosPrice']);
+    const userBalance = Number(formatUnits(userSystemTokenBalanceWei.value, WEI_PRECISION));
+    return (price * userBalance).toFixed(2);
+});
+
+// watchers
```
Comment (donnyquixotic) [inline src/components/header/AppHeaderWallet.vue:66]:
```suggestion
```

File: src/components/header/AppHeaderWallet.vue [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+watchEffect(() => {
+    if (isLoggedIn.value) {
+        fetchUserBalance();
+    }
+});
+
+// methods
```
Comment (donnyquixotic) [inline src/components/header/AppHeaderWallet.vue:73]:
```suggestion
```

File: src/lib/chain-utils.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    return Number(process.env.NETWORK_EVM_CHAIN_ID) === TELOS_MAINNET_CHAIN_ID;
+}
+
+export function isTestnet() {
+    return [TELOS_TESTNET_CHAIN_ID].includes(Number(process.env.NETWORK_EVM_CHAIN_ID));
+}
+
+export function isTelosTestnet() {
+    return Number(process.env.NETWORK_EVM_CHAIN_ID) === TELOS_TESTNET_CHAIN_ID;
+}
```
Comment (donnyquixotic) [inline src/lib/chain-utils.ts:21]: Redundant network check methods. `isMainnet` isn't used. Let's remove the telos-specfic methods and replace with `isMainnet` and `isTestnet` for now.

**Comment by donnyquixotic** [comment]
@ezra-sg great job on this! minor feedback

**Review by donnyquixotic — APPROVED** [review]

File: .eslintrc.js [hunk 6]
```diff
@@ -36,7 +36,6 @@ module.exports = {
         'comma-dangle': ['error', 'always-multiline'],
         'indent': ['error', 4],
         'quotes': ['error', 'single'],
-        'max-len': ['error', { 'code': 120 }],
```
Comment (Viterbo) [inline .eslintrc.js:39]: THANK YOU 🙏

File: src/components/header/AppHeaderSearch.vue [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    searchTerm.value = searchTerm.value.trim().replace(/\s/, '');
+    if (searchTerm.value.startsWith('0x')) {
+        if (searchTerm.value.length === 42) {
+            $router.push(`/address/${searchTerm.value}`);
+            return;
+        } else {
+            $router.push(`/tx/${searchTerm.value}`);
+            return;
+        }
+    } else if (searchTerm.value.match(/(^[a-z1-5.]{1,11}[a-z1-5]$)|(^[a-z1-5.]{12}[a-j1-5]$)/)) {
```
Comment (Viterbo) [inline src/components/header/AppHeaderSearch.vue:33]: This RegEx is complex enough to deserve its own commented explanation

(That's another way to say I didn't get a thing of what it says there lmao 🤣 )

**Review by Viterbo — APPROVED** [review]
Dude, you rock!

File: src/components/header/AppHeaderSearch.vue (hunk 7 above)
Comment (ezra-sg) [reply src/components/header/AppHeaderSearch.vue:33]: Yeah that was there when I joined telos haha. I believe it checks if the string is a valid telos native account name... maybe I can verify that and leave a comment

**Comment by pmjanus** [comment]
Having issues with network select links, they dont lead anywhere for me on desktop from local or https://deploy-preview-549--teloscan-redesign.netlify.app/ using pc chrome.

**Pull request merged**